java/dynamodb-tools/src/main/java/sleeper/dynamodb/tools/DynamoDBUtils.java

```diff
@@ -73,4 +77,22 @@ public static void configureTimeToLive(AmazonDynamoDB dynamoDB, String tableName
                 ));
         LOGGER.info("Configured TTL on field {}", expiryField);
     }
+
+    public static List<Map<String, AttributeValue>> doScanWithPagination(AmazonDynamoDB dynamoDB, ScanRequest scanRequest) {
+        double totalCapacity = 0.0D;
+        ScanResult result = dynamoDB.scan(scanRequest);
+        if (null != result.getConsumedCapacity()) {
+            totalCapacity += result.getConsumedCapacity().getCapacityUnits();
+        }
+        List<Map<String, AttributeValue>> allItems = result.getItems();
+        while (null != result.getLastEvaluatedKey()) {
+            result = dynamoDB.scan(scanRequest.withExclusiveStartKey(result.getLastEvaluatedKey()));
+            if (null != result.getConsumedCapacity()) {
+                totalCapacity += result.getConsumedCapacity().getCapacityUnits();
+            }
+            allItems.addAll(result.getItems());
+        }
+        LOGGER.debug("Scanned for records, capacity consumed = {}", totalCapacity);
```

Comment:
It might not be worth logging capacity consumption. This log message also won't say which call used the capacity. If you returned a stream of ScanResult, the caller could decide what to do with it?